src/savegame_json.cpp

```diff
@@ -859,6 +859,18 @@ void npc::load(JsonObject &data)
         attitude = npc_attitude(atttmp);
     }
 
+    if ( data.read( "companion_mission", comp_miss) ) {
```

Comment:
Why are you loading this into a temporary object? This `if` statement could be replaced by loading it directly:
```C++
data.read( "companion_mission", companion_mission );
```
If the JSON data does not contain the entry, the `read` function won't change the object.